sklearn/model_selection/_validation.py

```diff
@@ -393,9 +393,18 @@ def cross_val_predict(estimator, X, y=None, groups=None, cv=None, n_jobs=1,
         raise AttributeError('{} not implemented in estimator'
                              .format(method))
 
-    if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
-        le = LabelEncoder()
-        y = le.fit_transform(y)
+    do_manual_encoding = method in ['decision_function', 'predict_proba',
```

Comment:
I don't like this variable name. Perhaps just `encode`, `encoded` or `is_encoded` would do